packages/backend/src/helpers/compute-parameters.ts

```diff
@@ -1,13 +1,24 @@
 import type { IAction } from '@plumber/types'
 
 import get from 'lodash.get'
+import map from 'lodash/map'
 
 import ExecutionStep from '@/models/execution-step'
 
 import Step from '../models/step'
 
+const GET_ALL_SEPARATOR = ','
+
 const variableRegExp =
-  /({{step\.[\da-f]{8}-(?:[\da-f]{4}-){3}[\da-f]{12}(?:\.[\da-zA-Z-_ ]+)+}})/g
+  /({{step\.[\da-f]{8}-(?:[\da-f]{4}-){3}[\da-f]{12}(?:\.[\w* -]+)+}})/g
```

Comment:
can be slightly stricter to enforce that asterisk (*) must be the only character in the segment

valid: `{{step.00000000-0000-0000-0000-000000000000.*.abc}}`
invalid: `{{step.00000000-0000-0000-0000-000000000000.def*.abc}}`
```suggestion
  /({{step\.[\da-f]{8}-(?:[\da-f]{4}-){3}[\da-f]{12}(?:\.([\w -]+|\*))+}})/
```

Reply:
this does not enforce that asterisk cannot be in the last segment. for that the check can be done later.

Reply:
updated regex and added helper func to validate the step variable.
we only allow 1 wildcard and it cannot be at the start or end of the variable segment, example

invalid: `{{step.00000000-0000-0000-0000-000000000000.*.abc}}`
invalid: `{{step.00000000-0000-0000-0000-000000000000.abc.*}}`
invalid: `{{step.00000000-0000-0000-0000-000000000000.abc*.def.*.ghi}}`
valid: `{{step.00000000-0000-0000-0000-000000000000.abc.*.def}}`